ttnn/cpp/ttnn/operations/eltwise/binary_ng/device/kernels/dataflow/fill_tile_utils.hpp

```diff
@@ -0,0 +1,88 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "dataflow_api.h"
+
+// Fills one full tile of bfloat16 with a scalar value
+// Scalar is assumed to be a 16-bit value double packed into a u32
+FORCE_INLINE void fill_with_val(uint32_t cb_id, uint32_t packed_scalar) {
+    auto* ptr = reinterpret_cast<volatile tt_l1_ptr uint32_t*>(get_write_ptr(cb_id));
+    // 1024 is the number of elements in a full tile, but since the scalar is packed into a u32,
+    // each iteration writes 2 elements, hence the division by 2
+    for (uint32_t i = 0; i < 512; ++i) {
+        ptr[i] = packed_scalar;
+    }
+}
```

Comment:
I think we should make it explicit that `fill_with_val` is dtype specific, either with function name or dtype passed in and we `static_assert`